### ISSUE TITLE
ansible-lint reports use_ssh_args parameter of ansible.posix.synchronize as invalid

### ISSUE THREAD

**Issue by piotrp** [ansible/ansible-lint#3152]
ansible-lint 6.14.1 doesn't recognize `use_ssh_args` argument of `ansible.posix.synchronize` module (but recognizes nonexistent `ssh_args`).

```
$ ansible-lint --version
ansible-lint 6.14.1 using ansible 2.14.3
```
```
$ ansible-galaxy collection list

# /home/ppr/.ansible/collections/ansible_collections
Collection    Version
------------- -------
ansible.posix 1.5.1
```
```
$ cat synchronize-test.yml
- name: Test
  hosts: localhost
  tasks:
    - name: Copy files
      ansible.posix.synchronize:
        src: .
        dest: .
        owner: false
        group: false
        use_ssh_args: true
```
```
$ ansible-lint synchronize-test.yml
WARNING  Listing 1 violation(s) that are fatal
args[module]: Unsupported parameters for (basic.py) module: use_ssh_args. Supported parameters include: _local_rsync_password, _local_rsync_path, _substitute_controller, archive, checksum, compress, copy_links, delay_updates, delete, dest, dest_port, dirs, existing_only, group, link_dest, links, mode, owner, partial, perms, private_key, recursive, rsync_opts, rsync_path, rsync_timeout, set_remote_user, src, ssh_args, ssh_connection_multiplexing, times, verify_host. (warning)
synchronize-test.yml:4 Task/Handler: Copy files

Read documentation for instructions on how to ignore specific rule violations.

                  Rule Violation Summary
 count tag          profile rule associated tags
     1 args[module]         syntax, experimental (warning)

Passed with production profile, 5/5 star rating: 0 failure(s), 1 warning(s) on 1 files.
```

**Comment by ssbarnea** [comment]
I find impossible to reproduce this bug with current version of the linter. I even tried the same version of ansible,... quite weird.

**Comment by apatard** [comment]
I guess it's same kind of bug as in https://github.com/ansible/ansible-lint/issues/3110. use_ssh_args is only used by the action module synchronize.py and not by the module synchronize.py.

**Comment by piotrp** [comment]
I can replicate this in standalone Docker container, run using `docker run --rm -it -v //e/ansible-lint-test:/data python:3.10 bash` (mounted volume contains synchronize-test.yml):

```sh
$ cd /data
$ pip install 'ansible-core == 2.14.3' 'ansible-lint == 6.14.1'
$ ansible-galaxy collection install ansible.posix

$ ansible-galaxy collection list
Collection    Version
------------- -------
ansible.posix 1.5.1

$ ansible-lint --version
ansible-lint 6.14.1 using ansible 2.14.3

$ ansible-lint synchronize-test.yml
WARNING  Listing 1 violation(s) that are fatal
args[module]: Unsupported parameters for (basic.py) module: use_ssh_args. Supported parameters include: _local_rsync_password, _local_rsync_path, _substitute_controller, archive, checksum, compress, copy_links, delay_updates, delete, dest, dest_port, dirs, existing_only, group, link_dest, links, mode, owner, partial, perms, private_key, recursive, rsync_opts, rsync_path, rsync_timeout, set_remote_user, src, ssh_args, ssh_connection_multiplexing, times, verify_host. (warning)
synchronize-test.yml:4 Task/Handler: Copy files

Read documentation for instructions on how to ignore specific rule violations.

                  Rule Violation Summary
 count tag          profile rule associated tags
     1 args[module]         syntax, experimental (warning)

Passed with production profile, 5/5 star rating: 0 failure(s), 1 warning(s) on 1 files.
```

**Comment by felixfontein** [comment]
What ansible-lint should be doing is check whether an action plugin of that name is present, and if there is, it **must not check the module (if there is any) except if it is really, really sure that the action plugin only forwards to the module**. The current behavior (assuming that every action is a module) is just plain wrong.

It would probably better to use the options docs (`ansible-doc --json` output) to validate action plugins, and only use the module argument spec if there is no action plugin of that name.
